### PR TITLE
Add Value documentation for all functions

### LINKED ISSUE #871: Add `@return` value for all functions
Many functions are missing an `@return` value in their roxygen documentation. This should be added in ahead of the CRAN release.

### PR DESCRIPTION
**Pull request by edelarua**
Closes #871 

Also updated "Functions" section or "Notes" section for clarity where necessary. Applied the same structure for all statistics/formatted analysis/analyze function documentation.

E.g. `count_occurrences`:

example format

### CODE REVIEW

File: R/decorate_grob.R [hunk 1]
```diff
@@ -26,13 +26,7 @@
...
-#' The layout can be illustrated as follows:
-#'
-#' \figure{decorate_grob.png}{options: alt="decorate_grob layout"}
-#' }
-#'
-#' @return a grid grob (\code{gTree})
+#' @return A grid grob (`gTree`).
 #'
 #' @export
 #' @author Adrian Waddell (waddella) \email{adrian.waddell@roche.com}
```
Comment (Melkiades) [inline R/decorate_grob.R:32]: Is this email still existent? I think we should drop these in case

File: R/decorate_grob.R [hunk 2]
```diff
@@ -238,7 +232,7 @@ decorate_grob <- function(grob,
 }
 
 #' @importFrom grid validDetails
-#' @export
+#' @noRd
```
Comment (Melkiades) [inline R/decorate_grob.R:235]: Why these are no more exported?

Comment (Melkiades) [reply R/decorate_grob.R:235]: It seems they are not used a lot

File: R/abnormal_by_baseline.R [hunk 3]
```diff
@@ -19,12 +16,16 @@
...
+#' * `df` should be filtered to include only post-baseline records.
+#' * If the baseline variable or analysis variable contains `NA`, it is expected that `NA` has been
+#'   conveyed to `na_level` appropriately beforehand with [df_explicit_na()] or [explicit_na()].
+#'
 #' @inheritParams argument_convention
 #' @param abnormal (`character`)\cr identifying the abnormal range level(s) in `.var`.
 #' @seealso Relevant description function [d_count_abnormal_by_baseline()].
 #'
 #' @name abnormal_by_baseline
-#' @include formats.R
```
Comment (Melkiades) [inline R/abnormal_by_baseline.R:27]: What was this doing?

File: R/abnormal_by_marked.R [hunk 4]
```diff
@@ -3,18 +3,18 @@
...
 #'     with `Single, not last` and `Last or replicated` levels, respectively.
 #'   * For `Any`: Numerator is the number of patients with either single or
 #'     replicated marked abnormalities.
 #'
-#' @details Note that `Single, not last` and `Last or replicated` levels are
-#' mutually exclusive. If a patient has abnormalities that meet both the `Single, not last`
-#' and `Last or replicated` criteria, then the patient will be counted only under the `Last or replicated` category.
+#' Note that `Single, not last` and `Last or replicated` levels are mutually exclusive. If a patient has
+#' abnormalities that meet both the `Single, not last` and `Last or replicated` criteria, then the
+#' patient will be counted only under the `Last or replicated` category.
```
Comment (Melkiades) [inline R/abnormal_by_marked.R:17]: It is fine like this but it could go into a note

File: R/abnormal_by_worst_grade.R [hunk 5]
```diff
@@ -18,19 +18,19 @@
 #'     the correct denominators when building the layout as it is used to define row splitting.
 #'   * A toxicity grade variable (e.g. `GRADE_ANL`) where all negative values from
 #'     `ATOXGR` are replaced by their absolute values.
-#'   * Prior to tabulation, `df` must filtered to include only post-baseline records with worst grade flags.
+#'
+#' @note Prior to tabulation, `df` must filtered to include only post-baseline records with worst grade flags.
```
Comment (Melkiades) [inline R/abnormal_by_worst_grade.R:22]:
```suggestion
#' @note Prior to tabulation, `df` must be filtered to include only post-baseline records with worst grade flags.
```

File: R/decorate_grob.R (hunk 2 above)
Comment (edelarua) [reply R/decorate_grob.R:234]: I think these are just used internally. If we want to export them we will need to write full documentation for these functions.

File: R/abnormal_by_baseline.R (hunk 3 above)
Comment (edelarua) [reply R/abnormal_by_baseline.R:27]: To use the `tern` formatting functions they need to be collated prior to any other functions that use them. `abnormal.R` is the first file to be collated alphabetically and this `@include` is already present in that file, so this is just a duplicate.

File: R/combination_function.R [hunk 6]
```diff
@@ -9,6 +9,8 @@
 #' @param e2 (`CombinationFunction`)\cr right hand side of logical operator.
 #' @param x (`CombinationFunction`)\cr the function which should be negated.
 #'
+#' @return Returns a logical value indicating whether the left hand side of the equation equals the right hand side.
+#'
 #' @exportClass CombinationFunction
 #' @export CombinationFunction
 #'
```
Comment (Melkiades) [inline R/combination_function.R:16]: Is this combinationfunction actually useful? It seems that overloads stuff that can be done without this

Comment (Melkiades) [reply R/combination_function.R:16]: It is well used, nevermind

File: R/count_occurrences.R [hunk 7]
```diff
@@ -1,5 +1,7 @@
 #' Occurrence Counts
 #'
+#' @description `r lifecycle::badge("stable")`
+#'
 #' Functions for analyzing frequencies and fractions of occurrences for patients with occurrence
 #' data. Primary analysis variables are the dictionary terms. All occurrences are counted for total
 #' counts. Multiple occurrences within patient at the lowest term level displayed in the table are
```
Comment (Melkiades) [inline R/count_occurrences.R:7]: After this line there is a note that might be in notes

File: R/count_occurrences_by_grade.R [hunk 8]
```diff
@@ -271,11 +287,16 @@ count_occurrences_by_grade <- function(lyt,
   )
 }
 
-#' @describeIn count_occurrences_by_grade Layout creating function which adds content rows using the
-#'   statistics function and additional format arguments (see below).
+#' @describeIn count_occurrences_by_grade Layout-creating function which can take content function arguments
+#'   and additional format arguments. This function is a wrapper for [rtables::summarize_row_groups()].
```
Comment (Melkiades) [inline R/count_occurrences_by_grade.R:291]: Amazing!! this is so much clearer now

File: R/cox_regression.R [hunk 9]
```diff
@@ -186,7 +196,7 @@ tidy.coxreg.univar <- function(x, # nolint
   result
 }
 
-#' @describeIn tidy_coxreg Custom tidy method for a Multi-variable Cox Regression
+#' @describeIn tidy_coxreg Custom tidy method for a multi-variable Cox regression.
```
Comment (Melkiades) [inline R/cox_regression.R:199]:
```suggestion
#' @describeIn tidy_coxreg Custom tidy method for a multivariate Cox regression.
```
I think this is better. I talked with Gabe regarding this a couple of weeks ago and he said that in statistics this is the only appropriate form.

File: R/coxph.R [hunk 10]
```diff
@@ -7,6 +7,8 @@
 #'
 #' @param x the variable for which pairwise result is expected
 #'
+#' @return Variable "paired".
```
Comment (Melkiades) [inline R/coxph.R:10]: This function is not used in tern it seems. I do not understand what it exactly should do tbh. deprecation?

Comment (Melkiades) [reply R/coxph.R:10]: It is also missing examples

File: R/coxph.R [hunk 11]
```diff
@@ -24,7 +26,9 @@ pairwise <- function(x) {
 #' The special term `univariate` indicate that the model should be fitted individually for
```
Comment (Melkiades) [inline R/coxph.R:26]: This is the same function as pairwise

**Review by Melkiades — APPROVED** [review]
Really Amazing work!! Thanks so much! This makes things MUCH clearer to understand and the underlying components are more evident now. This PR is probably fixing a couple more issues about structure that I filed months ago. We should check if also those are completed by this! 

Just see the comments and fix what you think is appropriate, then you can merge ;)

File: R/count_values.R [hunk 12]
```diff
@@ -4,15 +4,23 @@
...
+#'   and fails otherwise.
+#' * For `count_values()`, variable labels are shown when there is more than one element in `vars`,
+#'   otherwise they are hidden.
+#'
 #' @name count_values_funs
 #'
 NULL
 
-#' @describeIn count_values_funs Statistics Function which is a generic function to count values.
+#' @describeIn count_values_funs S3 generic statistics function to count values.
```
[the remark this answers is not in the document]
Comment (Melkiades) [reply R/count_values.R:17]: it is the same this

File: R/count_occurrences.R (hunk 7 above)
Comment (Melkiades) [reply R/count_occurrences.R:7]: You can fix this and many others by just search for "note" in R and you will see all the ones that are not in notes eheh

File: R/coxph.R (hunk 10 above)
Comment (edelarua) [reply R/coxph.R:10]: Created an issue for this: https://github.com/insightsengineering/tern/issues/899

**Pull request merged**